### PR TITLE
telephone_to method

### PR DESCRIPTION
**Pull request by BenMorganIO**
You might have heard that HTML5 added the `mailto` and `tel`. I was able to find a `mail_to` method inside of Rails, but there didn't seem to be anything for `telephone_to`.

In this PR, I took the time to provide a method to generate telephone links that will add pauses, optimize for pause separators/phone extensions, and even let you add a `skype: true` option.

See the examples above the method for what you can do with it [here](https://github.com/rails/rails/pull/17956/files#diff-26e33d6e883deb07332ac9c1922eadc0R482).

### CODE REVIEW

File: actionview/lib/action_view/helpers/url_helper.rb [hunk 1]
```diff
@@ -468,6 +468,64 @@ def mail_to(email_address, name = nil, html_options = {}, &block)
...
+      # * <tt>:pause_seperator</tt> - Instead of seperating extensions with 'ext', you can provide your own pause seperator such as 'x' or 'extension'.
+      #
+      # ==== Examples
+      #   telephone_to "16044401234"
+      #   # => <a href="tel:16044401234">16044401234</a>
+      #
+      #   telephone_to "604.440.1234", "(604)-440-1234"
+      #   # => <a href="tel:6044401234">604.440.1234</a>
+      #
+      #   telephone_to "(604)-440-1234 ext 123", "(604)-440-1234 ext 123"
```
Comment (chancancode) [inline actionview/lib/action_view/helpers/url_helper.rb]: These are also very North-American-certeric though

File: actionview/lib/action_view/helpers/url_helper.rb [hunk 2]
```diff
@@ -468,6 +468,64 @@ def mail_to(email_address, name = nil, html_options = {}, &block)
...
+      # +telephone_to+ has several methods for customizing the phone number itself by
+      # passing special keys to +html_options+.
+      #
+      # ==== Options
+      # * <tt>:pauses</tt> - Specify the extension or numbers to dial after a pause.
+      # * <tt>:skype</tt> - Instead of the +tel+ protocol, you can use +callto+ which is used by Skype.
+      # * <tt>:pause_seperator</tt> - Instead of seperating extensions with 'ext', you can provide your own pause seperator such as 'x' or 'extension'.
+      #
+      # ==== Examples
+      #   telephone_to "16044401234"
```
Comment (chancancode) [inline actionview/lib/action_view/helpers/url_helper.rb]: Are you talking about the generated URL or the link text?

**Comment by chancancode** [comment]
Sorry I kind of stalled this :speak_no_evil:

Do we have consensus what is the minimum required feature set for this? If the few of us can't agree on what's useful enough here, then it's probably a sign that this doesn't generalize very well and is better off leaving it to the app developers to implement themselves so they can control precisely how they would like things formatted and normalized etc.

**Comment by matthewd** [comment]
IMO, the biggest reason `mail_to` ever existed was the now-removed address-hiding functionality. It has enough benefits to justify its continued existence, but I wouldn't look at it as setting the new-link-helper bar.

After we eliminate the way-too-complex, we're left here with a thing here that's only minimally nicer than just using `link_to(.., 'tel:..')`.

Mostly, I'm just unconvinced that enough people would actually have use for this: a reasonable subset of developers will want a single `tel:` link for their contact phone number somewhere, sure... but if you're actually filling your application with phone links, I'm sure you're going to end up needing more control than the "easy/smart" single-arg version of this helper provides.

**Comment by BenMorganIO** [comment]
I'm thinking this helper might cause more trouble than, well, help. Therefore, I'm OK with this PR being rejected.

As per @matthewd mention towards its relationship with `link_to`, we could simply do:

```ruby
def tel_to(name = nil, options = nil, html_options = nil)
  link_to name, "tel:#{options}", html_options
end
```

We could simplify everything down to this if we just want to have all the new HTML protocols helpers, otherwise, lets close.

**Comment by chancancode** [comment]
I won't be using this much myself, so I can't really champion it. Let's close this for now, if someone want to give a compelling argument on a particular approach feel free to leave a comment, and we can discuss based on that.

Thanks everyone for the work!

**Pull request closed without merging**